charts/pulsar/templates/function-worker/function-worker-statefulset.yaml

```diff
@@ -0,0 +1,188 @@
+              until [ ${proxyServiceNumber} -ge 1 ]; do
+                echo "pulsar proxy {{ template "pulsar.fullname" . }} isn't initialized yet ... check in 10 seconds ...";
+                sleep 10;
+                proxyServiceNumber="$(nslookup -timeout=10 {{ template "pulsar.fullname" . }}-{{ .Values.proxy.component }} | grep Name | wc -l)";
+              done;
+      {{- with .Values.common.extraInitContainers }}
+      {{ toYaml . | indent 6 }}
+      {{- end }}
+      containers:
+        - name: "{{ template "pulsar.fullname" . }}-{{ .Values.functions.component }}"
+          image: "{{ .Values.images.functions.repository }}:{{ .Values.images.functions.tag }}"
+          imagePullPolicy: {{ .Values.images.functions.pullPolicy }}
+          {{- if .Values.functions.probe.liveness.enabled }}
+          livenessProbe:
+            httpGet:
+              path: /initialized
+              port: {{ .Values.functions.ports.http }}
+            initialDelaySeconds: {{ .Values.functions.probe.liveness.initialDelaySeconds }}
+            periodSeconds: {{ .Values.functions.probe.liveness.periodSeconds }}
+            failureThreshold: {{ .Values.functions.probe.liveness.failureThreshold }}
+          {{- end }}
+          {{- if .Values.functions.probe.readiness.enabled }}
+          readinessProbe:
+            httpGet:
+              path: /initialized
+              port: {{ .Values.functions.ports.http }}
+            initialDelaySeconds: {{ .Values.functions.probe.readiness.initialDelaySeconds }}
+            periodSeconds: {{ .Values.functions.probe.readiness.periodSeconds }}
+            failureThreshold: {{ .Values.functions.probe.readiness.failureThreshold }}
+          {{- end }}
+          {{- if .Values.functions.resources }}
+          resources:
+{{ toYaml .Values.functions.resources | indent 12 }}
+          {{- end }}
+          command: ["sh", "-c"]
+          args:
+            - >
+              bin/pulsar functions-worker;
+          ports:
+            # prometheus needs to access /metrics endpoint
+            - name: http
+              containerPort: {{ .Values.functions.ports.http }}
+            {{- if and .Values.tls.enabled .Values.tls.broker.enabled }}
+            - name: https
+              containerPort: {{ .Values.functions.ports.https }}
+            {{- end }}
+          env:
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+          volumeMounts:
+          {{- include "pulsar.function.worker.config.volumeMounts" . | nindent 10 }}
+{{- with .Values.functions.extraVolumeMounts }}
+{{ toYaml . | indent 10 }}
+{{- end }}
+      volumes:
+      {{- include "pulsar.function.worker.config.volumes" . | nindent 6 }}
+{{- with .Values.functions.extraVolumes }}
+{{ toYaml . | indent 6 }}
+{{- end }}
+  {{- end }}
```

Comment:
Seems two extra white spaces here?

Reply:
this won't cause syntax error, but fixed